src/main/java/com/auth0/utils/tokens/PublicKeyProvider.java

```diff
@@ -0,0 +1,39 @@
+package com.auth0.utils.tokens;
+
+import com.auth0.exception.PublicKeyException;
+
+import java.security.interfaces.RSAPublicKey;
+
+/**
+ * The interface to obtain a public key. This is used to configure signature verification for tokens signed
+ * with the RS256 asymmetric signing algorithm.
+ * Developers should provide an implementation of this interface when verifying a RS256 ID token.
+ *
+ * <p>The following example demonstrates using the {@code JwkProviderBuilder} from the
+ * <a href="https://github.com/auth0/jwks-rsa-java">jwks-rsa-java library</a> to fetch the public key.</p>
+ *
+ * <pre>
+ * SignatureVerifier sigVerifier = SignatureVerifier.forRS256(new PublicKeyProvider() {
+ *     &#064;Override
+ *     public RSAPublicKey getPublicKeyById(String keyId) throws PublicKeyException {
+ *         try {
+ *             Jwk jwk = new JwkProviderBuilder("https://your-domain.auth0.com/").build().get(keyId);
```

Comment:
Not the best example if the default cache logic won't be in use because of the instance being created every time. I'd move the provider outside 👍

Reply:
Great catch. Updated the example to move the provider outside.